nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java

```diff
@@ -177,74 +174,60 @@ private String getValue(String propertyName) {
-                        converter = entry.getValue();
-                        break;
-                    }
+        // search for a matching raw type
+        for (Entry<Type, Converter> entry : converters.entrySet()) {
+            Type type1 = entry.getKey();
+            if (type1 instanceof ParameterizedType) {
+                ParameterizedType ptype = (ParameterizedType)type1;
+                if (ptype.getRawType().equals(propertyType)) {
+                    return entry.getValue();
                 }
             }
         }
         
-        if (converter == null) {
-            // see if a common sense converter can be created
-            boolean commonSense = true;
-            try {
-                Method m = propertyType.getMethod("of", String.class);
-                converter = (Converter<T>) new CommonSenseConverter(m);
-                converters.put(propertyType, converter);
-                return converter;
-            } catch (NoSuchMethodException | SecurityException ex) {
-                try {
-                    Method m = propertyType.getMethod("valueOf",String.class);
-                    converter = (Converter<T>) new CommonSenseConverter(m);
-                    converters.put(propertyType, converter);
-                    return converter;
-                } catch (NoSuchMethodException | SecurityException ex1) {
-                    try {
-                        Constructor c = propertyType.getConstructor(String.class);
-                        converter = (Converter<T>) new CommonSenseConverter(c);
-                        converters.put(propertyType, converter);
-                        return converter;
-                    } catch (NoSuchMethodException | SecurityException ex2) {
-                        try {
-                            Method m = propertyType.getMethod("parse", CharSequence.class);
-                            converter = (Converter<T>) new CommonSenseConverter(m);
-                            converters.put(propertyType, converter);
-                            return converter;
-                        } catch (NoSuchMethodException | SecurityException ex3) {
-                            commonSense = false;
-                        }
-                    }
-                }
-            }
+        // see if a common sense converter can be created
+        Optional<Converter<T>> commonSenseConverter = CommonSenseConverter.forClass(propertyType);
+        if (commonSenseConverter.isPresent()) {
+            converters.put(propertyType, commonSenseConverter.get());
+            return commonSenseConverter.get();
         }
-        if (converter == null) {
-            throw new IllegalArgumentException("Unable to convert value to type " + propertyType.getCanonicalName());
+
+        throw new IllegalArgumentException("Unable to convert value to type " + propertyType.getCanonicalName());
+    }
+
+
+
+    private static <T> Class<T> boxedTypeOf(Class<T> type) {
```

Comment:
The spec says (https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/converters.asciidoc): 
> The following Converters are provided by MicroProfile-Config by default:

Then naming primitive and their wrapper types. Does this mean this code is actually build into some shared code or does it want to tell us how we have to implement this? If this is already implemented there should not be a need to have this method since primitives and their wrappers are already covered.

Reply:
The latter. We need to implement this, which is exactly why this conversion happens. Other option would be to populate default converters map with both options, which is likely even simpler.

Reply:
Unless... overrides. And of course there can be.